### ISSUE TITLE
Investigate if the built-in CNI driver in EKS can be deleted (and update docs accordingly)

### ISSUE THREAD

**Issue by errordeveloper** [cilium/cilium#10420]
Right now we instruct the user to disable built-in VPC CNI driver by running the following command:

```
kubectl -n kube-system set image daemonset/aws-node aws-node=docker.io/spaster/alpine-sleep
```

This is an effective solution, however conceptually it may seem a little odd to the user. Also, if user follows EKS upgrade instructions without knowing that this was done at some stage earlier, they will put the VPC CNI driver back and break Cilium.

**Comment by errordeveloper** [comment]
I have checked https://github.com/cilium/cilium/pull/10126 and https://github.com/cilium/cilium/pull/8831, but don't see a pointers to any findings that lead to taking this approach. @joestringer said on Slack that this was done because EKS was restoring the daemonset after deletion.

I have created a new 1.14 cluster, and ran `kubectl delete ds -n kube-system aws-node`. After 14h there is no sign of the daemonset being restored.
